app/controllers/api/vms_controller.rb

```diff
@@ -351,7 +351,7 @@ def vm_event(vm, event_type, event_message, event_time)
     def retire_vm(vm, id, data)
       desc = "#{vm_ident(vm)} retiring"
       desc << " on #{data['date']}" if Hash(data)['date'].present?
-      retire_resource(:vms, id, data)
+      method(:retire_resource).super_method.call(:vms, id, data)
```

Comment:
@abellotti are you OK if I kill https://github.com/ManageIQ/manageiq/pull/11034/commits/e670dceea084e1d2a2a9d62fddcfa826160044e4 and do the above in a follow up PR?